### PR TITLE
Restructure TrollMoves scripts

### PR DESCRIPTION
**Pull request by pnuu**
This PR restructures all the "main" scripts of trollmoves. Most of client/server/mirror were duplicate code, and here most of that is now moved to `trollmoves.move_it_base.py`. At the same time some Python 3 compatibility updates and example configuration files were added.

### CODE REVIEW

File: trollmoves/client.py [hunk 1]
```diff
@@ -49,7 +50,7 @@
 
 DEFAULT_REQ_TIMEOUT = 1
 
-HEARTBEAT_TOPIC = "/heartbeat/move_it_server"
+HEARTBEAT_TOPIC = "/heartbeat/move_it_client"
```
[the remark this answers is not in the document]
Comment (pnuu) [reply trollmoves/client.py]: This is the client, so thought it should say "client" also on the topic.

File: bin/move_it.py [hunk 2]
```diff
@@ -606,46 +598,52 @@ def fun(pathname):
     return tnotifier
 
 
-def terminate(chains):
-    for chain in chains.itervalues():
+def terminate(chainss):
```
[the remark this answers is not in the document]
Comment (pnuu) [reply bin/move_it.py]: Becaus conflict with the global `chains` variable.

File: trollmoves/client.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mraspaud) [reply trollmoves/client.py]: ok, but isn't this listening to a heartbeat from the server ?

File: bin/move_it.py [hunk 3]
```diff
@@ -392,31 +385,30 @@ def move_it(pathname, destinations, hook=None):
...
     """
-    e = None
+    err = None
     for dest in destinations:
-        LOGGER.debug("Copying to: " + dest)
+        LOGGER.debug("Copying to: %s", dest)
         dest_url = urlparse(dest)
         try:
             mover = MOVERS[dest_url.scheme]
         except KeyError, e:
```
[the remark this answers is not in the document]
Comment (pnuu) [reply bin/move_it.py]: Nope, found one other, too. Fixed.

**Comment by pnuu** [comment]
I haven't tested either, so don't know how much things are broken. Also agree on merging the other PRs first, and rebasing this after that is done.

File: trollmoves/client.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (pnuu) [reply trollmoves/client.py]: Ah. I clarified the variable name.

File: trollmoves/server.py [hunk 4]
```diff
@@ -765,7 +770,7 @@ class CTimer(Thread):
     t.cancel() # stop the timer's action if it's still waiting
     """
 
-    def __init__(self, interval, function, args=[], kwargs={}):
+    def __init__(self, interval, function, *args, **kwargs):
```
Comment (mraspaud) [inline trollmoves/server.py]: This change is not correct, args has to be passed as a list, kwargs as a dict as per the Thread call signature

Comment (pnuu) [reply trollmoves/server.py]: Oh, right. Thought those original versions were considered "dangerous" and didn't think of the use-case at all. Good catch!

**Comment by mraspaud** [comment]
As a note for future PRs, separating style cleaning/refactoring/bug fixes into different PRs would make it faster and easier to review :)

**Pull request merged**